fastapi/dependencies/utils.py

```diff
@@ -240,6 +244,20 @@ def is_scalar_sequence_field(field: ModelField) -> bool:
     return False
 
 
+def is_scalar_mapping_field(field: ModelField) -> bool:
+    if (field.shape in mapping_shapes) and not lenient_issubclass(
+        field.type_, BaseModel
+    ):
+        if field.sub_fields is not None:
```

Comment:
Reverse the condition and return early. [Do not move the requirements literally to the code.](https://szymonkrajewski.pl/why-should-you-return-early/)